jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -1077,6 +1079,23 @@ private HttpServletResponse getServletResponse(Response response)
         }
     }
 
+    private static String getIncludedPathInContext(HttpServletRequest request, boolean isPathInfoOnly)
+    {
+        String servletPath = isPathInfoOnly ? "/" : (String)request.getAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH);
```

Comment:
What if a user decides to add this attribute in their Servlet code?

Reply:
We ignore it. Our getAttribute always returns our values for these.

Reply:
In ee9/ee8 we have a wrapper on Request that forces these attributes from our code before it asks the actual stored attributes, I cannot find something similar in ee10.

But what about someone doing this ...

``` java
    public class ZedFilter implements Filter
    {
        @Override
        public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException
        {
            request.setAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH, "/alt-servlet-path");
            chain.doFilter(request, response);
        }
    }
```

And this filter is in the chain for DefaultServlet?

Reply:
Yes that will look like an include. to the default servlet.  But it has always been possible for users to do this, as they can wrap request and intercept the `getAttribute` method and we have had wrapper object identity up until 12, so we could not stop them doing that.
I think all the servlet API attributes are kind of vulnerable to being messed about with by bad applications, but I don't think they can do anything that way that they couldn't do otherwise.  Filter can just do a real include if they want.

Reply:
> It kind of feels like this attribute could be used to do an isIncluded, thus avoiding two attribute lookups? return null if it is not set.

Does make the a bit harder to follow but will reduce 1 call to `getAttribute()`. Pushed a commit for this now.